libs/dsb-client-gateway-secrets-engine/src/lib/service/azure-key-vault.service.ts

```diff
@@ -0,0 +1,236 @@
+import { Injectable, Logger, OnModuleInit } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import {
+  CertificateDetails,
+  PATHS,
+  SecretsEngineService,
+} from '../secrets-engine.interface';
+import { Span } from 'nestjs-otel';
+import { DefaultAzureCredential } from '@azure/identity';
+import { KeyVaultSecret, SecretClient } from '@azure/keyvault-secrets';
+import { setLogLevel } from '@azure/logger';
+setLogLevel('info');
+
+@Injectable()
+export class AzureKeyVaultService
+  extends SecretsEngineService
+  implements OnModuleInit
+{
+  private readonly logger = new Logger(AzureKeyVaultService.name);
+
+  protected client: SecretClient;
+  protected readonly prefix: string;
+
+  constructor(protected readonly configService: ConfigService) {
+    super();
+    this.prefix = this.configService.get('AZURE_KEY_PREFIX', 'ddhub/');
+  }
+
+  @Span('azure_kv_onModuleInit')
+  public async onModuleInit(): Promise<void> {
+    const vaultName = this.configService.get('AZURE_VAULT_NAME');
+    const url = `https://${vaultName}.vault.azure.net`;
```

Comment:
can we change it to get full url from .env variable?
I believe in case of enterprise solutions clients may have their own .vault.azure.net url